src/sysext/sysext.c

```diff
@@ -575,15 +575,13 @@ static int mount_overlayfs(
         if (noexec >= 0)
                 SET_FLAG(flags, MS_NOEXEC, noexec);
 
-        if (upper_dir) {
+        if (upper_dir && work_dir) {
                 r = append_overlayfs_path_option(&options, ",", "upperdir", upper_dir);
                 if (r < 0)
                         return r;
 
                 flags &= ~MS_RDONLY;
-        }
 
-        if (work_dir) {
```

Comment:
i am not sure how much i like this change. previously, if we'd pass one without the other the mount would fail, because overlayfs needs both.

But now if only one is specified you silently fail. which sucks.

or what am i missing?

why even, this change?

Reply:
We are passing either two paths or two NULLs. If we pass one but not the other, then it's a logic error somewhere. I'll add an assert that ensures that both are NULL or none.